vowpalwabbit/parse_args.cc

```diff
@@ -1272,98 +1275,130 @@ VW::LEARNER::base_learner* setup_base(options_i& options, vw& all)
 {
   auto setup_func = all.reduction_stack.top();
   all.reduction_stack.pop();
-  auto base = setup_func(options, all);
+  auto base = std::get<1>(setup_func)(options, all);
 
-  if (base == nullptr)
-    return setup_base(options, all);
+  // this mean that setup_func did not do any setup since it didnt add itself to the chain of learners
```

Comment:
nit:
```suggestion
  // this means that the current (base) reduction was not enabled (didn't do any setup) since it didn't add itself to the chain of learners
```

Reply:
I rephrased it once more, since the (base) is extra and added phrased it more explicitly in case you want to read it again